lib/LaTeXML/Core/Stomach.pm

```diff
@@ -147,6 +151,14 @@ sub invokeToken {
   my ($self, $token) = @_;
   no warnings 'recursion';
 INVOKE:
+  # If we are invoking a new token, we'll do a new read of arguments,
+  # and that should void any remaining CC_SMUGGLE_THE noexpands
+  if ($$token[1] == CC_SMUGGLE_THE) {
```

Comment:
Actually, if I just avoid this guard on `$token` but run the removal of the smuggle cc *every time* invokeToken is called, cleveref.sty loads correctly, _and_ all tests pass. So this may be the only code path that needs attention - in one sense.

However, it is a terrible idea from a pragmatic standpoint - this is a very hot path in the code, and gets called **11,572** times just for fully running an empty `\usepackage{cleveref}` doc. So one would prefer solving the smuggle issue either earlier, or only when obviously needed, so that we can stay cheap in runtime...

Reply:
With the guard on, as you see it in the diff here, it actually doesn't get called at all during a `\usepackage{cleveref}` but fixes our synthetic example from the PR description. So at least it's fast :>